homework-submission/todo-app/src/App.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from "react";
+import "./App.css";
+import Item from "./Item.js";
+import ItemData from "./ItemData.js";
+import Header from "./Header";
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      todos: ItemData
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(id) {
+    this.setState(prevState => {
+      const newState = prevState.todos.map(todo => {
+        if (todo.id === id) {
+          todo.done = !todo.done;
+        }
+        return todo;
+      });
+      return {
+        todos: newState
+      };
+    });
+  }
```

Comment:
The loop here is costly, because you are looping over all the data to find one item which you want to change.
and you have in line 31 a key in the Map loop, so you can pass this key as a prop to the < Post > component and use it to find the item